src/main/java/de/torui/coflsky/config/CoflConfig.java

```diff
@@ -0,0 +1,381 @@
+package de.torui.coflsky.config;
+
+import cc.polyfrost.oneconfig.config.Config;
+import cc.polyfrost.oneconfig.config.annotations.*;
+import cc.polyfrost.oneconfig.config.data.*;
+import de.torui.coflsky.CoflSky;
+
+/**
+ * Comprehensive OneConfig configuration for CoflSky.
+ * <p>
+ * This mirrors the settings structure maintained in the separate
+ * <code>Coding/Windsurf/Cofl-Oneconfig</code> project. Only the first half
+ * (General + Mod categories) is ported in this commit – remaining
+ * categories will be added subsequently to keep the diff readable.
+ */
+public class CoflConfig extends Config {
```

Comment:
Couldn't this whole file be generated from the 
Bytecode Generation (More Advanced, Less Common for this Specific Use Case):
You could use a bytecode generation library (like ASM, ByteBuddy, or cglib) to literally create a new CoflConfig class from scratch in memory at runtime, including adding fields and even applying annotations if needed.